feat: Allow Users to import Firebase configs from google-services.json

khaled-0 opened the pull request:
Refactor: Removed some casts , Reformatted some null checks, replaced View.OnClickListener with lambda, added a Javadoc, overall cleaned up a bit.
And
**New Feature:**
Now Users Can Import Configs From ``google-services.json``
Clicking on Firebase from Library Manager now leads to Here
![image](https://user-images.githubusercontent.com/74397286/145932920-2de980f2-b373-4c76-87f3-9ed346ce16d4.png)
Here's a preview of how it would look overall when importing from json.
Sorry for the orientation, idk bandicam wont record horizontally
https://cdn.discordapp.com/attachments/827528331721048084/920161244705611796/8mb.video-IY3-WTc1MWuM.mp4

khaled-0 committed:
feat: Allow Users to import Firebase configs from google-services.json

Refactor: Removed some casts , Reformatted some null checks, replaced View.OnClickListener with lambda, added a Javadoc, overall cleaned up a bit.
And
**New Feature:**
Now Users Can Import Configs From ``google-services.json``
Clicking on Firebase from Library Manager now leads to Here
[image]
Here's a preview of how it would look
[Video]

khaled-0 committed:
fix: Added null check before parsing google-services.json

idk, it might happen. who knows

hasrat-ali reviewed (commented):
I would not use xml for the import button, it has to be created programmatically.

hasrat-ali reviewed (commented):
I would not use xml for the import button, it has to be created programmatically.

khaled-0 commented:
> I would not use xml for the import button, it has to be created programmatically.

Sir Using XML Here Seemed Easy and Better.
Also theres no need of any Resource ID so its all good.

hasrat-ali commented:
> > I would not use xml for the import button, it has to be created programmatically.
> 
> Sir Using XML Here Seemed Easy and Better. Also theres no need of any Resource ID so its all good.

Yeah, I know that, but creating programeticly is easy and take less effort.

khaled-0 commented on app/src/main/java/com/besome/sketch/editor/manage/library/firebase/ManageFirebaseActivity.java:
```diff
@@ -218,25 +246,119 @@ private void toFirebaseActivity() {
+            hasNullConfig = true;
+        }
+        if (rtdbMatcher.find()) {
+            configureImportFirebaseConfigFromJson(realtime_db, rtdbMatcher.group(1));
```
TODO: Replace https:// & / from the url

khaled-0 committed:
fix: Remove https:// from firebase database url

There's never a / at the end and I don't think firebase would ever use http either.

DerGoogler commented:
> I would not use xml for the import button, it has to be created programmatically. 

Using XMLs is higher quality, my opinion. 🧐

ilyassesalama commented:
Damn, been almost 6 months, merge when?

Arashvscode commented:
> Damn, been almost 6 months, merge when?

Never 😂😂

JavkhlanK commented:
I feel like this should actually parse the JSON into an object, not as any other string using regular expressions.

bydani2329 commented:
View on github

On Sun, 8 May 2022, 04:09 Javkhlan K., ***@***.***> wrote:

> I feel like this should actually parse the JSON into an object, not as any
> other string using regular expressions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Sketchware-Pro/Sketchware-Pro/pull/285#issuecomment-1120280285>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOFF2TLFZ7QJSDY3AKE5EUTVI3EYLANCNFSM5J74NRSA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

khaled-0 commented:
> View on github
> […](#)
> On Sun, 8 May 2022, 04:09 Javkhlan K., ***@***.***> wrote: I feel like this should actually parse the JSON into an object, not as any other string using regular expressions. — Reply to this email directly, view it on GitHub <[#285 (comment)](https://github.com/Sketchware-Pro/Sketchware-Pro/pull/285#issuecomment-1120280285)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOFF2TLFZ7QJSDY3AKE5EUTVI3EYLANCNFSM5J74NRSA> . You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

That would be great too but for now , regex works fine

khaled-0 commented:
that's a lot of commits

JavkhlanK commented:
quite

khaled-0 committed:
add missing bracket, add correct imports

how in the world

khaled-0 committed:
Merge branch 'main' into google-services.joe

khaled-0 committed:
fix bugs occurred during merge

click listeners

khaled-0 commented:
Signed by *Joe Biden*

JavkhlanK commented on app/src/main/java/com/besome/sketch/editor/manage/library/firebase/ManageFirebaseActivity.java:
```diff
@@ -127,15 +142,47 @@ public void onClick(View view) {
+                SketchwareUtil.toastError("Please Configure Your Firebase Project Details First," +
+                        " Either Using Settings Icon Or Importing From google-services.json");
             }
+        switch (view.getId()) {
```
Why are you checking for the view ID again even though it's R.id.layout_switch?

khaled-0 replied:
Hels , I'm 100% sure I've removed these

khaled-0 replied:
Happened while merging

khaled-0 replied:
Anyway just remove the switch

JavkhlanK replied:
ok lol

zoropalingditakutizoro-netizen reviewed (commented):
CONTOH
